qiita_db/handlers/processing_job.py

```diff
@@ -175,26 +170,19 @@ def post(self, job_id):
         with qdb.sql_connection.TRN:
             job, success, error_msg = _get_job(job_id)
             if success:
-                if job.status != 'running':
-                    success = False
-                    error_msg = 'Job in a non-running state.'
-                else:
-                    payload = loads(self.request.body)
-                    if payload['success']:
-                        for artifact_data in payload['artifacts']:
-                            filepaths = artifact_data['filepaths']
-                            atype = artifact_data['artifact_type']
-                            parents = job.input_artifacts
-                            params = job.parameters
-                            qdb.artifact.Artifact.create(
-                                filepaths, atype, parents=parents,
-                                processing_parameters=params)
-                        job.status = 'success'
+                payload = loads(self.request.body)
+                try:
```

Comment:
Would you mind restricting the scope of this try block to just the line that might raise the exception?